modules/opencascade/opencascade/surface_shape.cpp

```diff
@@ -0,0 +1,65 @@
+#include "surface_shape.h"
+
+#include <opencascade/STEPCAFControl_Reader.hxx>
+#include <opencascade/TopExp_Explorer.hxx>
+#include <opencascade/TopoDS.hxx>
+#include <opencascade/BRep_Builder.hxx>
+#include <opencascade/GeomAPI_ProjectPointOnSurf.hxx>
+#include <opencascade/gp_Pnt.hxx>
+
+namespace SPH
+{
+	//=================================================================================================//
+Vecd SurfaceShape::findClosestPoint(const Vecd &input_pnt)
+{
+
+    Vecd closest_pnt;
+    gp_Pnt point1 = EigenToOcct(input_pnt);
+    gp_Pnt point2;
+    Extrema_ExtAlgo Algo = Extrema_ExtAlgo_Tree;
+    point2 = GeomAPI_ProjectPointOnSurf(point1, surface_, Algo);
+    closest_pnt[0] = point2.X();
+    closest_pnt[1] = point2.Y();
+    closest_pnt[2] = point2.Z();
+
+    return closest_pnt;
+}
+
+////=================================================================================================//
+bool SurfaceShape::checkContain(const Vecd &pnt, bool BOUNDARY_INCLUDED) { return 0; }
+BoundingBox SurfaceShape::findBounds() { return BoundingBox(); }
+//=================================================================================================//
+
+Vecd SurfaceShape::findActualPoint(Standard_Real u, Standard_Real v)
```

Comment:
findActualPoint, "Actual" seems confusing. Any better term?

Reply:
Also I would suggest to rename `find` too because the function only computes the cartesian coordinates from the parametric values. So something like `[compute or get][Cartesian or Spatial][Point or Coordinates or Coords]` might fit better